### PR TITLE
Create pipeline per branch instana-agent-operator in Concourse

### PR DESCRIPTION
**Pull request by Jefiya-MJ**
# Why
Instana-agent-operator runs PR tests with circle CI when the PR is open

# What

Move the pipeline from circle CI  to Concourse

# References

- [Kanbanize Card](https://instana.kanbanize.com/ctrl_board/50/cards/124466/details/)

### CODE REVIEW

File: ci/pr-pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  - name: self-update
+    max_in_flight: 1
+    plan:
+      - get: pipeline-source
+        trigger: true
+      - set_pipeline: self
+        file: pipeline-source/ci/pr-pipeline.yml
+        vars:
+          branch: *build-branch
+          project-berlin-gh-token: ((project-berlin-gh-token))
```
Comment (Milica-Cvrkota-IBM) [inline ci/pr-pipeline.yml:29]: Would it be possible to use this token for putting gh status? If we don't have to generate some new tokens and reuse this one, we can also implement putting the git status as part of this PR

File: ci/pr-pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            args:
+              - -ceu
+              - |
+                cd pipeline-source
+                microdnf install make python3-devel gcc git tar gzip zip curl jq
+                pip3 install pyyaml
+                
+                export PATH="$PATH:/usr/local/go/bin"
+                export GOPATH=/go
+                export GO_VERSION=1.19
```
Comment (Milica-Cvrkota-IBM) [inline ci/pr-pipeline.yml:91]: I think we want to use the same version for building and testing, we can use 1.18 for now to include in this PR only migration to Concourse

File: controllers/leaderelection/coordination_api/http_client.go [hunk 3]
```diff
@@ -74,7 +73,7 @@ func (c *podCoordinationHttpClient) PollPod(ctx context.Context, pod coreV1.Pod)
 			pod.GetObjectMeta().GetName(), resp.Status)
 	}
 
-	body, err := ioutil.ReadAll(resp.Body)
+	body, err := io.ReadAll(resp.Body)
```
Comment (Milica-Cvrkota-IBM) [inline controllers/leaderelection/coordination_api/http_client.go:76]: Was this change pushed by accident?

**Review by Milica-Cvrkota-IBM — CHANGES_REQUESTED** [review]

**Comment by Milica-Cvrkota-IBM** [comment]
We could also remove the Circle CI with this PR: https://github.com/instana/instana-agent-operator/tree/main/.circleci. I will see who can disable the pipeline on this repo within the CircleCI

File: controllers/leaderelection/coordination_api/http_client.go (hunk 3 above)
Comment (Jefiya-MJ) [reply controllers/leaderelection/coordination_api/http_client.go]: In Go 1.19, the io/ioutil package has been deprecated, and the functionality has been moved to io package

File: ci/pr-pipeline.yml (hunk 2 above)
Comment (Jefiya-MJ) [reply ci/pr-pipeline.yml]: In [unit test ](https://ci.instana.io/teams/agent/pipelines/jefiya-test/jobs/build-and-test-agent-operator/builds/3) build fails when we use version 1.18

```
/go/pkg/mod/go.uber.org/multierr@v1.10.0/error.go:224:20: undefined: atomic.Bool
note: module requires Go 1.19
```

**Review by Milica-Cvrkota-IBM — APPROVED** [review]

**Pull request merged**